tests/integration/components/upload-image-test.js

```diff
@@ -1,23 +1,55 @@
 import { module, test } from 'qunit';
 import { setupRenderingTest } from 'ember-qunit';
-import { render } from '@ember/test-helpers';
+import { render, find, triggerEvent, click } from '@ember/test-helpers';
 import { hbs } from 'ember-cli-htmlbars';
 
-module('Integration | Component | upload-image', function (hooks) {
+module('Integration | Component | image uploader', function (hooks) {
   setupRenderingTest(hooks);
-  test('it renders', async function (assert) {
+
+  const file = new File(['dummy image data'], 'test-image.jpg', {
+    type: 'image/jpeg',
+  });
+
+  test('it renders upload UI when no image is selected', async function (assert) {
     // Set upload url
-    this.set('uploadUrl', 'http://localhost:3000/test');
+    await render(hbs`
+    <UploadImage @uploadUrl={{this.uploadUrl}}  @formKeyName = {{this.formDataKeyName}}/>
+    `);
+    const button = this.element.querySelector('.image-form__button');
+
+    assert.dom('.drop-area').exists().hasNoClass('.drop-area__highlight');
+
+    // Test handling of file input change event
+    await triggerEvent('input[type="file"]', 'change', { files: [file] });
+
+    assert.dom('.drop-area').doesNotExist();
```

Comment:
Can we please use `data-test-selectors` here ? As the sole purpose of `data-test-selectors` is to grab elements for testing only while classes may be conflicting due to being used with various elements